core/state_processor.go

```diff
@@ -81,6 +81,9 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 		statedb.IntermediateRoot(true)
 	}
 
+	parentHeader := p.bc.GetHeaderByNumber(block.NumberU64() - 1)
```

Comment:
I'm not sure if this needs to be the parent header, but could be the current header.